tests/lib/rules/hook-use-state-names.js

```diff
@@ -0,0 +1,85 @@
+/**
+ * @fileoverview Ensure symmetric naming of setState hook value and setter variables
+ * @author Duncan Beevers
+ */
+
+'use strict';
+
+// ------------------------------------------------------------------------------
+// Requirements
+// ------------------------------------------------------------------------------
+
+const RuleTester = require('eslint').RuleTester;
+const rule = require('../../../lib/rules/hook-use-state-names');
+const parsers = require('../../helpers/parsers');
+
+// ------------------------------------------------------------------------------
+// Tests
+// ------------------------------------------------------------------------------
+
+const ruleTester = new RuleTester({
+  parserOptions: {
+    ecmaVersion: 2018,
+    sourceType: 'module'
+  }
+});
+
+ruleTester.run('hook-set-state-names', rule, {
+  valid: [
+    {
+      code: 'const [color, setColor] = useState()'
+    },
+    {
+      code: 'const [color, setColor] = useState(\'#ffffff\')'
+    },
+    {
+      code: 'const [color, setColor] = React.useState()'
+    },
+    {
+      code: 'const [color, setColor, unused] = useState()'
+    },
+    {
+      code: 'const [color1, setColor1] = useState()'
+    },
+    {
+      code: 'const [, setColor] = useState()'
+    },
+    {
+      code: 'const useStateResult = useState()'
+    },
+    {
+      code: 'const [, , unused] = useState()'
+    },
+    {
+      code: 'const [color, setColor] = useState<string>()',
+      parser: parsers.TYPESCRIPT_ESLINT
+    },
+    {
+      code: 'const [color, setColor] = useState<string>(\'#ffffff\')',
+      parser: parsers.TYPESCRIPT_ESLINT
+    }
```

Comment:
let's also duplicate these and add `parsers['@TYPESCRIPT_ESLINT']` so we can handle the new parser too. grep for `parsers.TS(` to find examples of how to do this safely.

Reply:
I think I added these correctly.